app/src/test/kotlin/compiler/parser/analysis/testcases/TestCase11.kt

```diff
@@ -0,0 +1,117 @@
+package compiler.parser.analysis.testcases
+
+import compiler.parser.analysis.GrammarAnalysis
+import compiler.parser.analysis.GrammarAnalysisTest.DfaFactory
+import compiler.parser.analysis.GrammarSymbol
+import compiler.parser.grammar.AutomatonGrammar
+import kotlin.test.Ignore
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class TestCase11 {
+
+    // COMPLICATED NON NULLABLE GRAMMAR (WITH PLUGGED TERMINAL)
+    // To test if we can detect t \notin FIRST(start) [t occurs after sth non trivially non nullable]
+    //
+    // Grammar:
+    //   Sigma = { start, Q, R, s, t }
+    //   Productions = {
+    //      start --> Q R R (start) + R t Q Q (dfaCNNG1)
+    //      Q --> s (dfaQ1)
+    //      R --> QQ + s (dfaR1)
+    //   }
+    //
+
+    private val start = "start"
+    private val symQ = "Q"
+    private val symR = "R"
+    private val syms = "s"
+    private val symt = "t"
+
+    private val expectedNullable = setOf<GrammarSymbol>()
+
+    private val expectedFirst = mapOf(
+        start to setOf(start, symQ, symR, syms),
+        symQ to setOf(symQ, syms),
+        symR to setOf(symQ, symR, syms),
+        syms to setOf(syms),
+        symt to setOf(symt),
+    )
+
+    private val expectedFollow: Map<GrammarSymbol, Set<GrammarSymbol>> = mapOf(
+        start to setOf(start, symQ, symR, syms, symt),
+        symQ to setOf(start, symQ, symR, syms, symt),
+        symR to setOf(start, symQ, symR, syms, symt),
+        syms to setOf(start, symQ, symR, syms, symt),
+        symt to setOf(start, symQ, syms),
```

Comment:
Start couldn't occur after `t` - please remove or add a comment if it's because of approximation

Reply:
youre right